### PR TITLE
External resource file loading and Interpret uncertainty #79 #94

### PR DESCRIPTION
**Pull request by qifeng-bai**
Make external file loading configurable 
https://github.com/AtlasOfLivingAustralia/la-pipelines/issues/79

Implement interpretation of coordinateUncertaintyInMeters #94

STATE_COORDINATE_MISMATCH logic incorrect #92

### CODE REVIEW

File: livingatlas/pipelines/pom.xml [hunk 1]
```diff
@@ -759,9 +759,9 @@
 
     <!-- Test -->
     <dependency>
-      <groupId>junit</groupId>
-      <artifactId>junit</artifactId>
-      <version>${junit4.version}</version>
+      <groupId>org.junit.jupiter</groupId>
+      <artifactId>junit-jupiter-api</artifactId>
+      <version>5.6.2</version>
```
Comment (djtfmartin) [inline livingatlas/pipelines/pom.xml:764]: Can you move this to a property ?

File: livingatlas/pipelines/src/test/java/au/org/ala/parser/DistanceRangeParserTest.java [hunk 2]
```diff
@@ -30,21 +32,19 @@ public void parseRange() {
     value = DistanceRangeParser.parse("100km,");
     assertEquals(100000f, value, 0);
 
-    value = DistanceRangeParser.parse(" 1km-20km,");
+    value = DistanceRangeParser.parse(" 1kilometers-20kilometres");
```
Comment (djtfmartin) [inline livingatlas/pipelines/src/test/java/au/org/ala/parser/DistanceRangeParserTest.java:35]: why has the original test been replaced ? shouldnt we just have both tests ?
Similarly for the other examples....

File: livingatlas/pipelines/src/test/java/au/org/ala/pipelines/interpreters/AlaLocationInterpreterTest.java [hunk 3]
```diff
@@ -135,9 +144,10 @@ public void assertionMaximumDistanceAboveSurfaceInMetersTest() {
     coreMap.put(DwcTerm.maximumDistanceAboveSurfaceInMeters.qualifiedName(), "2err0 inch");
     LocationInterpreter.interpretMaximumDistanceAboveSurfaceInMeters(er, lr);
 
-    assertEquals(Double.valueOf(0.51d), lr.getMaximumDistanceAboveSurfaceInMeters());
+    assertEquals(lr.getMaximumDistanceAboveSurfaceInMeters(), Double.valueOf(0.51d));
```
Comment (djtfmartin) [inline livingatlas/pipelines/src/test/java/au/org/ala/pipelines/interpreters/AlaLocationInterpreterTest.java:147]: Expected and Actual are the wrong way round.
See Error prone logs

File: livingatlas/pipelines/src/test/java/au/org/ala/pipelines/interpreters/AlaTemporalInterpreterTest.java [hunk 4]
```diff
@@ -27,8 +27,8 @@ public void testQualityAssertion() {
     TemporalRecord tr = TemporalRecord.newBuilder().setId("1").build();
     ALATemporalInterpreter.interpretTemporal(er, tr);
     assertArrayEquals(
-        new String[] {ALAOccurrenceIssue.MISSING_COLLECTION_DATE.name()},
-        tr.getIssues().getIssueList().toArray());
+        tr.getIssues().getIssueList().toArray(),
```
Comment (djtfmartin) [inline livingatlas/pipelines/src/test/java/au/org/ala/pipelines/interpreters/AlaTemporalInterpreterTest.java:30]: Expected and Actual are the wrong way round.
See Error prone logs

File: livingatlas/pipelines/src/test/java/au/org/ala/pipelines/interpreters/AlaTemporalInterpreterTest.java [hunk 5]
```diff
@@ -39,12 +39,12 @@ public void testQualityAssertion() {
 
     ALATemporalInterpreter.interpretTemporal(er, tr);
     assertArrayEquals(
+        tr.getIssues().getIssueList().toArray(),
```
Comment (djtfmartin) [inline livingatlas/pipelines/src/test/java/au/org/ala/pipelines/interpreters/AlaTemporalInterpreterTest.java:42]: Expected and Actual are the wrong way round.
See Error prone logs

File: livingatlas/pipelines/src/test/java/au/org/ala/pipelines/interpreters/AlaTemporalInterpreterTest.java [hunk 6]
```diff
@@ -104,8 +104,8 @@ public void testInvalidAssertions() {
     ALATemporalInterpreter.interpretTemporal(er, tr);
 
     assertArrayEquals(
-        new String[] {OccurrenceIssue.RECORDED_DATE_UNLIKELY.name()},
-        tr.getIssues().getIssueList().toArray());
+        tr.getIssues().getIssueList().toArray(),
```
Comment (djtfmartin) [inline livingatlas/pipelines/src/test/java/au/org/ala/pipelines/interpreters/AlaTemporalInterpreterTest.java:107]: Expected and Actual are the wrong way round.
See Error prone logs

File: livingatlas/pipelines/src/test/java/au/org/ala/pipelines/interpreters/AlaTemporalInterpreterTest.java [hunk 7]
```diff
@@ -86,11 +86,11 @@ public void testAUformatDatessertions() {
     ALATemporalInterpreter.interpretTemporal(er, tr);
 
     assertArrayEquals(
+        tr.getIssues().getIssueList().toArray(),
```
Comment (djtfmartin) [inline livingatlas/pipelines/src/test/java/au/org/ala/pipelines/interpreters/AlaTemporalInterpreterTest.java:89]: Expected and Actual are the wrong way round.
See Error prone logs

File: livingatlas/pipelines/src/test/java/au/org/ala/pipelines/interpreters/AlaTemporalInterpreterTest.java [hunk 8]
```diff
@@ -63,11 +63,11 @@ public void testALAssertions() {
...
```
Comment (djtfmartin) [inline livingatlas/pipelines/src/test/java/au/org/ala/pipelines/interpreters/AlaTemporalInterpreterTest.java:66]: Expected and Actual are the wrong way round.
See Error prone logs

File: livingatlas/pipelines/src/test/java/au/org/ala/pipelines/interpreters/AlaTemporalInterpreterTest.java (hunk 6 above)
Comment (qifeng-bai) [reply livingatlas/pipelines/src/test/java/au/org/ala/pipelines/interpreters/AlaTemporalInterpreterTest.java]: Checked others and pushed

File: livingatlas/pipelines/src/test/java/au/org/ala/parser/DistanceRangeParserTest.java (hunk 2 above)
Comment (qifeng-bai) [reply livingatlas/pipelines/src/test/java/au/org/ala/parser/DistanceRangeParserTest.java:38]: Since there is a test for km, but no harm to add it back

**Pull request merged**